pydantic/networks.py

```diff
@@ -350,6 +397,98 @@ class PostgresDsn(AnyUrl):
     }
     user_required = True
 
+    __slots__ = AnyUrl.__slots__ + ('hosts',)
+
+    def __init__(self, *args, hosts: Optional[List['HostParts']] = None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.hosts = hosts
+
+    @classmethod
+    def validate_host_parts(cls, parts: 'HostParts') -> 'HostParts':
+        """
+        A method used to validate host parts of an URL.
+        """
+        port = parts['port']
+        if port is not None and int(port) > 65_535:
+            raise errors.UrlPortError()
+
+        return parts
+
+    @classmethod
+    def validate_multi_host(cls, hosts: List[str]) -> List['HostParts']:
+        hosts_parts: List['HostParts'] = []
+        for host in hosts:
+            hm = host_regex().match(host)
+            original_parts = cast('Parts', hm.groupdict())
+            host, tld, host_type, rebuild = cls.validate_host(original_parts)
+            host_parts = cast(
+                'HostParts',
+                {
+                    'host': host,
+                    'host_type': host_type,
+                    'tld': tld,
+                    'rebuild': rebuild,
+                    'port': original_parts.get('port'),
+                },
+            )
+            host_parts = cls.validate_host_parts(host_parts)
+            hosts_parts.append(host_parts)
+        return hosts_parts
+
+    @classmethod
+    def validate(cls, value: Any, field: 'ModelField', config: 'BaseConfig') -> 'PostgresDsn':
```

Comment:
There's still to much duplication here.

There are a number of ways to work around this while maintaining performance and not altering the external interface to `AnyUrl`.

This class should only override `validate_host`, or perhaps a new method `validate_netloc`.

Reply:
This part is the most tricky.

It is hard to override it since we need to build a new objects with new field `hosts`. We could map `host` and other fields and join them by `,` but the problem would be that we will lost which port relies to which host ect.

I will think more what would be the best way to merge it.

@samuelcolvin Please let me know if you have already something in mind.